odoo/tools/func.py

```diff
@@ -77,19 +89,22 @@ def filter_kwargs(func, kwargs):
 
     return {key: kwargs[key] for key in kwargs if key not in leftovers}
 
-def synchronized(lock_attr='_lock'):
+
+def synchronized(lock_attr: str = '_lock'):
```

Comment:
`typing.Optional` ? for python 3.10 ?

Reply:
Optional does not mean it is an optional argument to the function; the argument has a default value. `Optional[X] = X | None`, here we always want a string.